src/helpers/translations/index.ts

```diff
@@ -1,12 +1,31 @@
 import {
   init as initFbt,
+  IntlVariations,
   IntlViewerContext,
 } from 'fbt';
 import translations from '../../i18n/translatedFbts.json';
 
+import continuationLocalStorage from 'cls-hooked';
+
 const DEFAULT_LOCALE = 'en_US';
 
-export async function loadTranslations(locale: string) {
-  IntlViewerContext.locale = locale || DEFAULT_LOCALE;
-  initFbt({ translations });
+export async function loadTranslations(locale: string, isNode?: boolean) {
+  initFbt({
+    translations,
+    hooks: {
+      getViewerContext() {
+        if (!!isNode && continuationLocalStorage != null) {
+          const sessionLocale = continuationLocalStorage.getNamespace('session').get('locale');
+          if (!sessionLocale) {
+            throw new Error('Unknown sessionLocale');
+          }
+          locale = sessionLocale;
+        }
+        return {
+          GENDER: IntlVariations.GENDER_UNKNOWN,
+          locale: locale || DEFAULT_LOCALE
+        };
+      }
```

Comment:
Depends on a new Fbt runtime API from this PR: https://github.com/facebook/fbt/pull/142